app/cdap/components/NamespaceAdmin/SourceControlManagement/index.tsx

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import PrimaryContainedButton from 'components/shared/Buttons/PrimaryContainedButton';
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import T from 'i18n-react';
+import Table from 'components/shared/Table';
+import TableHeader from 'components/shared/Table/TableHeader';
+import TableRow from 'components/shared/Table/TableRow';
+import TableCell from 'components/shared/Table/TableCell';
+import TableBody from 'components/shared/Table/TableBody';
+import { useSelector, useDispatch } from 'react-redux';
+import ActionsPopover from 'components/shared/ActionsPopover';
+import { UnlinkSourceControlModal } from './UnlinkSourceControlModal';
+import { NamespaceAdminActions } from '../store';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import { MyNamespaceApi } from 'api/namespace';
+import { MySecureKeyApi } from 'api/securekey';
+import { ISourceControlManagement } from 'components/NamespaceAdmin/SourceControlManagement/types';
+import { Observable } from 'rxjs/Observable';
+import { SourceControlManagementModal } from './SourceControlManagementModal';
+
+const PREFIX = 'features.SourceControlManagement';
+
+const StyledInfo = styled.div`
+  font-weight: 400;
+  font-size: 13px;
+  color: #bdbdbd;
+  margin-bottom: 15px;
+`;
+
+export const SourceControlManagement = () => {
+  const params = {
+    namespace: getCurrentNamespace(),
+  };
+  const [isFormOpen, setIsFormOpen] = useState(false);
+  const [isUnlinkModalOpen, setIsUnlinkModalOpen] = useState(false);
+  const sourceControlManagement = useSelector((state) => state.sourceControlManagement);
+  const dispatch = useDispatch();
+  const toggleForm = () => {
+    setIsFormOpen(!isFormOpen);
+  };
+  const toggleUnlinkModal = () => {
+    setIsUnlinkModalOpen(!isUnlinkModalOpen);
+  };
+
+  const getBodyForSubmit = (formState, validate = false) => {
+    return {
+      test: validate,
+      repository: formState,
+    };
+  };
+
+  const getSourceControlManagement = () => {
```

Comment:
I don't think these functions should be in this component

Reply:
I think we need to come to a consensus on how we want to do api requests - @njbriggs do we have any examples of doing this properly? these should really be pulled out to a separate file using a redux thunk (pretty standard way of doing it before) or using custom hooks to accomplish this.

Reply:
Also, please use `pipe` when chaining Observables

Reply:
moved api calls to actioncreater, and added a pipe. Please check if it's the correct usage of pipe